plugin/src/main/java/net/dzikoysk/funnyguilds/rank/RankSystem.java

```diff
@@ -1,9 +1,75 @@
 package net.dzikoysk.funnyguilds.rank;
 
-public enum RankSystem {
+import com.google.common.collect.ImmutableMap;
+import net.dzikoysk.funnyguilds.FunnyGuilds;
+import net.dzikoysk.funnyguilds.config.IntegerRange;
+import net.dzikoysk.funnyguilds.config.PluginConfiguration;
 
-    ELO,
-    PERCENT,
-    STATIC
+import java.util.function.BiFunction;
+
+public class RankSystem {
+
+    private ImmutableMap<RankType, BiFunction<Integer, Integer, RankResult>> map;
+
+    private RankSystem() {
+    }
+
+    public RankResult calculate(RankType type, Integer attackerPoints, Integer victimPoints) {
+        return map.get(type).apply(attackerPoints, victimPoints);
+    }
+
+    public enum RankType {
+        ELO,
+        PERCENT,
+        STATIC,
+    }
+
+    public static class RankResult {
+
+        private final Integer attackerPoints;
+        private final Integer victimPoints;
+
+        public RankResult(Integer attackerPoints, Integer victimPoints) {
+            this.attackerPoints = attackerPoints;
+            this.victimPoints = victimPoints;
+        }
+
+        public RankResult(Integer samePoints) {
+            this.attackerPoints = samePoints;
+            this.victimPoints = samePoints;
+        }
+
+        public Integer getAttackerPoints() {
+            return attackerPoints;
+        }
+
+        public Integer getVictimPoints() {
+            return victimPoints;
+        }
+    }
+
+    public static RankSystem create() {
+        RankSystem rankSystem = new RankSystem();
+        PluginConfiguration config = FunnyGuilds.getInstance().getPluginConfiguration();
+
+        rankSystem.map = new ImmutableMap.Builder<RankType, BiFunction<Integer, Integer, RankResult>>()
```

Comment:
Czemu to nie może być po prostu przekazywane przez konstruktor i oznaczone jako finalny field

Reply:
Bo ja to pewnie bede musial zmienic moim GH-1653 ( ͡° ͜ʖ ͡°)

Reply:
w sumie mądre